src/MathOptInterface.jl

```diff
@@ -94,6 +94,36 @@ Empty the model, that is, remove all variables, constraints and model attributes
 """
 function empty! end
 
+"""
+    supports_incremental_interface(model::ModelLike, copy_names::Bool)
+
+Return a `Bool` indicating whether `model` supports building incrementally via
+[`add_variable`](@ref) and [`add_constraint`](@ref).
+
+`copy_names` is a `Bool` indicating whether the user wishes to set
+[`VariableName`](@ref) and [`ConstraintName`](@ref) attributes.
+If `model` supports the incremental interface but does not support name
+attributes, define
+```julia
+supports_incremental_interface(::MyNewModel, copy_names::Bool) = !copy_names
+```
+
+The main purpose of this function is to determine whether a model can be loaded
+into `model` incrementally or whether it should be cached and copied at once
+instead.
+
+This is used by [`instantiate`](@ref) to determine whether to add a cache in two
+situations:
+1. A first cache can be used to store the model as entered by the user as well
+   as the names of variables and constraints. This cache is created if this
```

Comment:
This cache is not added by `instantiate`, it is added by JuMP independently of `supports_incremental_interface`

Reply:
https://github.com/jump-dev/MathOptInterface.jl/blob/db801f5981d61c01f05ba2fce249bc55d4750e86/src/instantiate.jl#L115-L130

Reply:
That's the cache added by 2., it stores the bridged model, not the model entered by the user. Instantiate never creates 2 caches. JuMP does but the first cache is not done by `instantiate`